File: Anton and Danik-CF734-D2-A.cpp
#include <iostream>
#include "cmath"
using namespace std;
 
 
int main() {
 
    int n;
    string s;
    cin>>n>>s;
    int A=0;
    int D=0;
    for(int i=0;i<s.size();i++)
    {
        if(s[i]=='A')
            A++;
        else
            D++;
    }
 
    if(A>D)
        cout<<"Anton"<<endl;
    else if (D>A)
        cout<<"Danik"<<endl;
    else
        cout<<"Friendship"<<endl;
 
 
 
 
 
 
    return 0;
}

File: Anton and Danik-CF734-D2-A.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
 
        Scanner input = new Scanner(System.in);
      
       int n =input.nextInt();
        String winner = input.next();
        int A=0 , D=0;
 
        for(int i=0;i<winner.length(); i++)
        {
            if(winner.charAt(i)=='A')
                A++;
            else
                D++;
        }
 
        if (A>D)
            System.out.println("Anton");
        else if (D>A)
            System.out.println("Danik");
        else
            System.out.println("Friendship");
    }
}
